test/index.test.js

```diff
@@ -1,9 +1,94 @@
 'use strict';
 
 const assert = require('chai').assert;
+const mockery = require('mockery');
+const sinon = require('sinon');
 
-describe('index test', () => {
-    it('fails', () => {
-        assert.isTrue(false);
+describe('Template Publish', () => {
+    let requestMock;
+    let YamlMock;
+    let fsMock;
+    let index;
+    let yamlReturn;
+
+    before(() => {
+        mockery.enable({
+            warnOnUnregistered: false,
+            useCleanCache: true
+        });
+    });
+
+    beforeEach(() => {
+        requestMock = sinon.stub();
+        YamlMock = {
+            safeLoad: sinon.stub()
+        };
+        fsMock = {
+            readFileSync: sinon.stub()
+        };
+        yamlReturn = {
+            name: 'template/test',
+            version: '1.0.0',
+            description: 'Publishes the template yaml from sd-template.yaml',
+            maintainer: 'tiffanykyi@gmail.com',
+            config: {
+                image: 'node:6',
+                steps: [
+                    { publish: 'node ./publish.js' }
+                ]
+            }
+        };
+
+        YamlMock.safeLoad.returns(yamlReturn);
+
+        mockery.registerMock('fs', fsMock);
+        mockery.registerMock('js-yaml', YamlMock);
+        mockery.registerMock('request', requestMock);
+
+        /* eslint-disable global-require */
+        index = require('../index');
+        /* eslint-enable global-require */
+    });
+
+    afterEach(() => {
+        mockery.deregisterAll();
+        mockery.resetCache();
+        index = null;
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('throws error when request yields an error', () => {
+        const error = new Error('error');
+
+        requestMock.yields(error);
+        assert.throws(index.publishTemplate, Error, 'Error sending request: Error: error');
```

Comment:
It's an asynchronous operation, so any errors it throws won't be catch-able. If you switch the mock over to `.yieldsAsync`, this error will show up.